src/main/scala/co/triptailor/sandbox/ClassificationService.scala

```diff
@@ -33,19 +33,32 @@ trait ClassificationService { self: Common =>
     (for {
       doc  ← m
       dl   = compute_dl(doc)
-    } yield ClassifiedDocument(doc, rating = compute_bm25(doc, tags, dl, avdl))).sorted
+      ratedTags = rateTags(doc.metrics, tags, dl, avdl)
+    } yield ClassifiedDocument(doc, rating = compute_bm25(ratedTags, tags, dl, avdl), ratedTags)).sorted
   }
 
-  private def compute_bm25(doc: RatedDocument, tags: Seq[String], dl: Double, avdl: Double): Double =
-    tags.flatMap(doc.metrics.get).foldLeft( 0d ) { (rating, metrics) =>
-      rating + (metrics.cfreq * metrics.sentiment) / (metrics.freq * (1 - b + b * (dl / avdl)))
+  private def compute_bm25(ratedTags: Seq[RatedTag], tags: Seq[String], dl: Double, avdl: Double): Double =
+    ratedTags.foldLeft( 0d ) { (rating, token) =>
+      rating + token.rating
     }
+  
+  private def rateTags(metrics: Map[String, RatingMetrics], tags: Seq[String], dl: Double, avdl: Double) = {
+    val filteredMetrics = getMetricsOfTags(tags, metrics)
+    filteredMetrics.map { token =>
+      val tokenStr = token._1
```

Comment:
Maybe we should rename `tokenStr` to `tag` to be consistent with our naming conventions